src/bulkUseCaseRequest.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  isTestBoxBulkUseCaseRequest,
+  ITestBoxTrial,
+  ITestBoxBulkUseCaseRequest,
+  UseCaseType,
+} from "./payloads";
+import axios from "axios";
+import { TestBoxError } from "./error";
+import { verifyAuthenticationToken } from "./auth";
+import { Request, Response } from "express";
+import { getConfigItem } from "./config";
+import TestBoxTrial from "./trial";
+
+type UseCaseUrls = { [key: string]: string }
+
+export default class TestBoxBulkUseCaseRequest implements ITestBoxBulkUseCaseRequest {
+  version: 1;
+  trial_id: string;
+  use_case_types: UseCaseType[];
+  trial_data: ITestBoxTrial;
+  success_url: string;
+  failure_url: string;
+
+  private hasVerifiedAuth: boolean = false;
+  private authToken: string;
+
+  constructor(payload: ITestBoxBulkUseCaseRequest) {
+    // The constructor only checks that the contract between the SDK and TestBox
+    // is being upheld. It does not check for authorization/authentication,
+    // as this is done async.
+    if (isTestBoxBulkUseCaseRequest(payload)) {
+      this.version = payload.version;
+      this.trial_id = payload.trial_id;
+      this.use_case_types = payload.use_case_types;
+      this.trial_data = new TestBoxTrial(payload.trial_data);
+      this.failure_url = payload.failure_url;
+      this.success_url = payload.success_url;
+    } else {
+      throw new TestBoxError(
+        "An invalid use case payload was provided to the TestBoxBulkUseCaseRequest class."
+      );
+    }
+  }
+
+  static async fromExpressRequest(req: Request) {
+    let payload = req.body;
+    if (typeof req.body === "string") {
+      payload = JSON.parse(req.body);
+    }
+    const trialRequest = new TestBoxBulkUseCaseRequest(payload);
+    await trialRequest.verifyToken(
+      req.headers.authorization.replace("Bearer ", "")
+    );
+    return trialRequest;
+  }
+
+  async verifyToken(authToken: string): Promise<boolean> {
```

Comment:
🧠 Might want to consider refactoring the `verifyToken` and `throwIfAuthNotValidated` into a new base class, `TestBoxRequest` or something, and then have `TestBoxBulkUseCaseRequest` and `TestBoxTrialRequest` inherit from it.